features/steps/cantabularfeature.go

```diff
@@ -0,0 +1,72 @@
+package steps
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular"
+	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular/gql"
+	"github.com/ONSdigital/dp-cantabular-filter-flex-api/config"
+	"github.com/ONSdigital/dp-cantabular-filter-flex-api/features/mock"
+	"github.com/ONSdigital/dp-cantabular-filter-flex-api/service"
+
+	"github.com/cucumber/godog"
+)
+
+type CantabularFeature struct {
+	*mock.CantabularClient
+}
+
+func NewCantabularFeature() *CantabularFeature {
+	cf := &CantabularFeature{CantabularClient: &mock.CantabularClient{OptionsHappy: true}}
+	service.GetCantabularClient = func(cfg *config.Config) service.CantabularClient {
```

Comment:
I feel like the New() should just be returning a new instance of whatever struct, not changing things in the service package. This should probably be done elsewhere

Reply:
from what I can see the injection of the external dependencies is done either in the component (as at present) or the feature itself can do it?